File: Java_programs_Eclipse/src/learn_practice/armstrong.java
package learn_practice;

import java.util.Scanner;
import java.lang.Math;

public class armstrong {

	public static void main(String[] args) {
		Scanner read = new Scanner(System.in);
		
		System.out.print("Enter integer : ");
		int a = read.nextInt();
		int b = a;
		int c = a;
		
		int temp = 0;
		int count = 0;
		int sum = 0;
		
		while(b != 0)
		{
			temp = b / 10;
			count++;
			b = b / 10;
		}
		System.out.println("Number of digits in a integer : " + count);
		
		while(c != 0)
		{
			int s = c % 10;
			int r = (int)Math.pow(s,count);
			sum = sum + r;
			c = c / 10;
		}
		System.out.println(sum);
		
		if(a == sum)
		{
			System.out.println("It is an armstrong number");
		}
		else
		{
			System.out.println("It is not an armstrong number");
		}
		
	}
}

File: C_ program_VScode/armstrong.c
#include<stdio.h>
#include<math.h>

int main()
{
    int n,r,sum=0,temp,c = 0,count=0,d;
    int tempn;
    printf("Enter a no.: ");
    scanf("%d", &n);
    temp = n;
    tempn = n;
    while(tempn != 0)
    {
        tempn = tempn /10;
        count++;
    }
    printf("the no. of digits are %d\n",count);
  
    while(n>0)
    {
        r = 0;
        r = n % 10;
        c = 0;
        c =  round(pow(r, count));                                     //pow(4 , 2)   
         printf("c = %d\n",c);

        sum = sum + c;
        n = n / 10;
    }

    printf("%d\n",temp);
    printf("%d\n",sum);

    if(temp == sum)
    printf("It is a armstrong no.");
    else
    printf("It is not a armstrong");
}